app/js/controllers/settings.js

```diff
@@ -5,21 +5,39 @@
 export default class SettingsController extends Controller {
   constructor(options) {
     super(options);
+
+    this.merging = false;
+
+    this._refreshAddons = this._refreshAddons.bind(this);
```

Comment:
I find saving state like this to be a bit ugly and more error-prone. Let's write the code below like this instead:
```js
navigator.mozApps.mgmt.removeEventListener(name, (e) => this._refreshAddons(e));
```